kpi/deployment_backends/base_backend.py

```diff
@@ -1,30 +1,65 @@
 # coding: utf-8
+import copy
 import json
+from typing import Union
 
-from bson import json_util, ObjectId
+from bson import json_util
+from django.db import transaction
+from django.utils import timezone
 from django.utils.translation import ugettext_lazy as _
 from rest_framework import serializers
 from rest_framework.pagination import _positive_int as positive_int
 
 from kpi.constants import INSTANCE_FORMAT_TYPE_XML, INSTANCE_FORMAT_TYPE_JSON
+from kpi.exceptions import AbstractMethodError
+from kpi.utils.jsonbfield_helper import ReplaceValues
 
 
 class BaseDeploymentBackend:
 
-    # TODO. Stop using protected property `_deployment_data`.
-
     INSTANCE_ID_FIELDNAME = '_id'
+    STATUS_SYNCED = 'synced'
+    STATUS_NOT_SYNCED = 'not-synced'
 
     def __init__(self, asset):
         self.asset = asset
         # Python-only attribute used by `kpi.views.v2.data.DataViewSet.list()`
         self.current_submissions_count = 0
 
+    def get_data(self,
+                 dotted_path: str = None,
+                 default=None) -> Union[None, int, str, dict]:
+        """
+        Access `self.asset._deployment_data` and return corresponding value of
+        `dotted_path` if it exists. Otherwise, it returns `default`.
+        If `dotted_path` is not provided, it returns the whole
+        dictionary.
+        """
+        if not dotted_path:
+            # We do not want to return the mutable object whose could be altered
+            # later. `self.asset._deployment_data` should never be accessed
+            # directly
+            return copy.deepcopy(self.asset._deployment_data) # noqa
```

Comment:
if we're going to be très pédant, how about 2 spaces before the comment, eh? :stuck_out_tongue: